twitter_scrooge/twitter_scrooge.bzl

```diff
@@ -257,6 +257,10 @@ def _compile_generated_scala(
         label.name + "_scalac.statsfile",
         sibling = scrooge_jar,
     )
+    diagnosticsfile = ctx.actions.declare_file(
```

Comment:
maybe add it to the outputs of scrooge?

Reply:
I may be interpreting this wrong but isn't the outputs of scrooge the same as the others rules? This eventually calls `compile_scala` which runs the action

Reply:
Maybe I'm confused. You only added it to scalapb and scrooge. Why? My guess is because this line solves it for the other rules https://github.com/bazelbuild/rules_scala/pull/1068/files#diff-f57e74eda2d86b1794ed00d7d9359137R8
If I'm right then maybe we can do the same for these rules

Reply:
Ah yes I get it now! You're right. Should each of these rules have its own common outputs? Also, it would seem it generates also some of the other common outputs, such as the `statsfile` and the `MANIFEST`. Should those live in such a file as well?

Reply:
I believe that can be done but perhaps a different name should be chosen since the `statsfile` and `MANIFEST` are in no way related to diagnostics. Do you know what they stand for specifically?

Reply:
Statsfile is also for diagnostics (compilation times).
We can just add it to each of the outputs for now.

Reply:
After trying to implement this, it would seem that this is not exactly viable since both scrooge and scalapb are aspects not rules, and they can't have outputs associated to them. Is this correct? I'm still fairly new to all of this so please correct me if I'm wrong

Reply:
I think you're right. Let's keep it as is